pkgs/development/libraries/rocm-runtime/rocr-ext-dir.diff

```diff
@@ -0,0 +1,24 @@
+diff --git a/src/core/util/lnx/os_linux.cpp b/src/core/util/lnx/os_linux.cpp
+index fdbe19a..42d4ef8 100644
+--- a/src/core/util/lnx/os_linux.cpp
++++ b/src/core/util/lnx/os_linux.cpp
+@@ -161,8 +161,17 @@ static_assert(sizeof(Mutex) == sizeof(pthread_mutex_t*), "OS abstraction size mi
+ static_assert(sizeof(Thread) == sizeof(os_thread*), "OS abstraction size mismatch");
+ 
+ LibHandle LoadLib(std::string filename) {
+-  void* ret = dlopen(filename.c_str(), RTLD_LAZY);
+-  if (ret == nullptr) debug_print("LoadLib(%s) failed: %s\n", filename.c_str(), dlerror());
++  std::string extDirFilename = GetEnvVar("ROCR_EXT_DIR") + "/" + filename;
++  void* ret = dlopen(extDirFilename.c_str(), RTLD_LAZY);
```

Comment:
First, is there a downside to calling `dlopen` like this without checking the result of the `GetEnvVar` call?

Second, this logic will also be applied for every call to to `LoadLib`, so it seems like it will affect things when loading the aqlprofile library. I do not know if that is the interface used by AMD's current profiling tools as that's another area where things are up in the air, but the way this change impacts every path passed to `LoadLib` strikes me as a potential danger for introducing incompatibility.

Reply:
I can move the handling to the call site and add a check when `GetEnvVar` is empty.

Reply:
I apologize for not thinking about this more carefully with the earlier review.

Reply:
No problem. Somehow I overlooked that too. First made a PR here:

https://github.com/NixOS/nixpkgs/pull/93388